### PR TITLE
First Java example not working

### PR DESCRIPTION
**Pull request by rgugliel**
First Java example does not work with recent Spark version (see https://issues.apache.org/jira/browse/SPARK-19156)

*Make sure that you generate site HTML with `jekyll build`, and include the changes to the HTML in your pull request also. See README.md for more information. Please remove this message.*

### CODE REVIEW

File: examples.md [hunk 1]
```diff
@@ -413,4 +413,4 @@ model.transform(df).show();
 Many additional examples are distributed with Spark:
 
  * Basic Spark: [Scala examples](https://github.com/apache/spark/tree/master/examples/src/main/scala/org/apache/spark/examples), [Java examples](https://github.com/apache/spark/tree/master/examples/src/main/java/org/apache/spark/examples), [Python examples](https://github.com/apache/spark/tree/master/examples/src/main/python)
- * Spark Streaming: [Scala examples](https://github.com/apache/spark/tree/master/examples/src/main/scala/org/apache/spark/examples/streaming), [Java examples](https://github.com/apache/spark/tree/master/examples/src/main/java/org/apache/spark/examples/streaming)
\ No newline at end of file
+ * Spark Streaming: [Scala examples](https://github.com/apache/spark/tree/master/examples/src/main/scala/org/apache/spark/examples/streaming), [Java examples](https://github.com/apache/spark/tree/master/examples/src/main/java/org/apache/spark/examples/streaming)
```
Comment (srowen) [inline examples.md:416]: What changed here, newline? if you can revert it.

File: examples.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ counts.saveAsTextFile("hdfs://...")
 {% highlight java %}
 JavaRDD<String> textFile = sc.textFile("hdfs://...");
 JavaRDD<String> words = textFile.flatMap(new FlatMapFunction<String, String>() {
-  public Iterable<String> call(String s) { return Arrays.asList(s.split(" ")); }
+  public Iterator<String> call(String s) { return Arrays.asList(s.split(" ")).iterator(); }
```
Comment (srowen) [inline examples.md:65]: You need to regenerate site HTML too. See README.md

**Review by srowen — CHANGES_REQUESTED** [review]
Please improve the title too

File: examples.md (hunk 1 above)
Comment (rgugliel) [reply examples.md:416]: I'm not sure and I don't know how to revert this change.

File: examples.md (hunk 2 above)
Comment (rgugliel) [reply examples.md:65]: Done, and pushed site/examples.html

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
@rgugliel I squashed and merge this. You will need to manually close it because the merged commit is different.

**Comment by rgugliel** [comment]
Thanks!

**Pull request closed without merging**